### PR TITLE
Add diff log to `release`

### PR DESCRIPTION
**Pull request by dylanpyle**
Test output from https://github.com/dylanpyle/cala-release-scratchpad :

```
$ npx cala-release minor
gpg: WARNING: unsafe permissions on homedir '/Users/dylan/.gnupg'
From github.com:dylanpyle/cala-release-scratchpad
 * branch            main       -> FETCH_HEAD
Already up to date.
v1.2.0
Enumerating objects: 6, done.
Counting objects: 100% (6/6), done.
Delta compression using up to 8 threads
Compressing objects: 100% (4/4), done.
Writing objects: 100% (4/4), 1.04 KiB | 1.04 MiB/s, done.
Total 4 (delta 2), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To github.com:dylanpyle/cala-release-scratchpad.git
   038ab69..9b255ac  main -> main
 * [new tag]         v1.2.0 -> v1.2.0
Switched to and reset branch 'production'
Your branch is ahead of 'origin/production' by 2 commits.
  (use "git push" to publish your local commits)
From github.com:dylanpyle/cala-release-scratchpad
 * branch            production -> FETCH_HEAD
Already up to date.
Already up to date.
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
To github.com:dylanpyle/cala-release-scratchpad.git
   5f91dd1..9b255ac  production -> production
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
ℹ️ Changes in this release (please share a changelog with the team):
commit 038ab69b7dbfd44946d546a962a4d60dedc81728
Author: Dylan Pyle <me@dylanpyle.com>
Date:   Tue Aug 17 13:37:12 2021 -0400

    new file
```

### CODE REVIEW

**Comment by dylanpyle** [comment]
Screen Shot 2021-08-17 at 13 39 31

File: release [hunk 1]
```diff
@@ -50,6 +50,15 @@ run() {
...
 
   git checkout main
+
+  local recent_tags=$(git for-each-ref --sort=-creatordate --format '%(refname)' refs/tags | head -n 2)
+  local previous_tag=$(echo "$recent_tags" | tail -n 1)
+  local this_tag=$(echo "$recent_tags" | head -n 1)
+
+  if [[ "$previous_tag" != '' && $this_tag != '' ]]; then
+    echo "ℹ️ Changes in this release (please share a changelog with the team):"
+    git log $previous_tag..$this_tag^
```
Comment (scotttrinh) [inline release]: Sorry for the drive-by review, but I wonder if `--oneline` would lead to more readable output since it wouldn't include the author, etc?

Comment (dylanpyle) [reply release]: @scotttrinh I thought about it but then you lose the commit body... then I was briefly playing with custom formatting but got a bit 🐰 🕳️ ...

It's interesting to see everyone's different approaches to PR squash merge messages; it looks like yours usually don't include a body (in which case `--oneline` would definitely make more sense!) whereas I usually copy the PR description into the squashed commit (probably why I leaned towards this) - everyone else is somewhere in the middle.

so... i'm not thoroughly convinced one way or the other is "better"  but I wonder if having to do the changelog might encourage us to make those commit bodies more descriptive in the first place? 🤷

Comment (scotttrinh) [reply release]: Yeah, that's interesting. I guess my only feeling is that whatever this outputs, is exactly what we should copy into whatever place it needs to be copied into without any editing, so this seems fine.

As far as whether commit message bodies are something that we should care about, I'd love to have more discussion here! I find that so much of the context of the change happens in the communication in the PR itself, so it's hard to rely on the PR description alone to get the gist of the change, but knowing that they're externally facing might help guide how we write these descriptions.

Comment (dylanpyle) [reply release]: I've thought about it a bit more and I'm cool with a short one

`--oneline` also includes less useful stuff like the hash, and doesn't include author/date, so i dipped my toe in the 🐰 🕳️  a tiny bit and came up with 1e6b94d — let me know what you think

sample output:
```
- 2021-08-17 (Dylan Pyle) Shorter log
- 2021-08-17 (Dylan Pyle) Add diff log to `release`
- 2020-07-01 (Dylan Pyle) Fix package.json
```

**Review by scotttrinh — APPROVED** [review]
💅
